.github/workflows/githubci.yml

```diff
@@ -0,0 +1,32 @@
+name: Arduino-timer Library CI
+
+on: [pull_request, push, repository_dispatch]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    
+    steps:
+    - uses: actions/setup-python@v1
+      with:
+        python-version: '3.x'
+    - uses: actions/checkout@v2
+    - uses: actions/checkout@v2
+      with:
+         repository: adafruit/ci-arduino
+         path: ci
+
+    - name: pre-install
+      run: bash ci/actions_install.sh
+
+    - name: test platforms
+      #run: python3 ci/build_platform.py main_platforms
+      run: python3 ci/build_platform.py main_platforms esp32 esp8266
+      #run: python3 ci/build_platform.py main_platforms esp32 esp8266 nrf52840
+
+    # disabled for now.
+    #- name: doxygen
+      #env:
+      #  GH_REPO_TOKEN: ${{ secrets.GH_REPO_TOKEN }}
+      #  PRETTYNAME : "arduino-timer Library"
+      #run: bash ci/doxy_gen_and_deploy.sh
```

Comment:
Uncommenting this section runs doxygen, then attempts to push its output files to a